R/create_custom_check.R

```diff
@@ -0,0 +1,113 @@
+#' Create a custom validation check function template file.
+#'
+#' @param name Character string. Name of the custom check function. We recommend
+#' following the hubValidations package naming convention. For more details, consult the
+#' article on [writing custom check functions](
+#' https://hubverse-org.github.io/hubValidations/articles/writing-custom-fns.html).
+#' @param hub_path Character string. Path to the hub directory. Default is the
+#' current working directory.
+#' @param r_dir Character string. Path (relative to `hub_path`) to the directory
+#' the custom check function file will be written to. Default is `src/validations/R`
+#' which is the recommended directory for storing custom check functions.
+#' @param error Logical. If `TRUE`, the custom check function will return a
+#' `<error/check_error>` class object in the case of a failed check instead of the
+#' default `<error/check_failure>`.
+#' @param conditional Logical. If `TRUE`, the custom check function template will
+#' include a block of code to check a condition before running the check. This is useful
+#' when a check may need to be skipped based on a condition.
+#' @param error_object Logical. If `TRUE`, the custom check function template will
+#' include an error object that can be used to store additional information about the
+#' properties of the object being checked that caused check failure. For example,
+#' it could store the index of rows in a `tbl` that caused a check failure.
+#' @param extra_args Logical. If `TRUE`, the custom check function template will
+#' include an `extra_arg` template argument and template block of code to check
+#' the input arguments of the custom check function.
+#' @param overwrite Logical. If `TRUE`, the function will overwrite an existing
+#'
+#' @return Invisible `TRUE` if the custom check function file is created successfully.
+#' @details
+#' See the article on [writing custom check functions](
+#' https://hubverse-org.github.io/hubValidations/articles/writing-custom-fns.html)
```

Comment:
Note the article URL results in a 404 currently as it is still under development but will work in the next PR